db/migrate/20150106050733_set_mysql_to_unicode_mb4.rb

```diff
@@ -0,0 +1,43 @@
+class SetMysqlToUnicodeMb4 < ActiveRecord::Migration
+  # Converts the tables and strings columns to utf8mb4, which is the true, full
+  # unicode support in MySQl
+  UTF8_PAIRS = {}
+  tables = ActiveRecord::Base.connection.tables
+
+  tables.each do |table|
+    ActiveRecord::Base.connection.columns(table).each do |column|
+      # build a hash with all the columns that contain text
+      if (column.type == :string) || (column.type == :text)
+        UTF8_PAIRS[table] = { :name => column.name, :type => column.sql_type }
```

Comment:
This is not good. We should have the key the column. This is a serious error from my side, sorry.